packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -773,43 +773,44 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
   }
 
   /**
-   *  {@inheritDoc ISavedObjectsRepository.bulkDelete}
+   * performs initial checks on object type validity and flags multi-namespace objects for preflight checks
+   * @param objects SavedObjectsBulkDeleteObject[]
+   * @returns array BulkDeleteExpectedBulkGetResult[]: left (error) or right Either result
    */
-  async bulkDelete(
-    objects: SavedObjectsBulkDeleteObject[],
-    options: SavedObjectsBulkDeleteOptions = {}
-  ): Promise<SavedObjectsBulkDeleteResponse> {
-    const { refresh = DEFAULT_REFRESH_SETTING, force } = options;
-    const namespace = normalizeNamespace(options.namespace);
+  private presortObjectsByNamespaceType(objects: SavedObjectsBulkDeleteObject[]) {
```

Comment:
@rudolf @pgayvallet There are pros and cons of moving these `bulkDelete` utility methods to their own module, the main ones of which are:
- Pro: reduce the file size a bit
- Con: higher risk of reusing them elsewhere
Would it be okay to keep them as private within the repository to speed this work up a bit?